### DISCUSSION TITLE
Downloaded video shorter than original

### DISCUSSION THREAD

**Discussion by Gutdzilla** [aclap-dev/video-downloadhelper#113] [Help and Q&A] [answered]
### Your question

Hi!

When a video is downloaded by VDH it is shorter than original file.
The time is shorter in different length. I download file with original length equal 11:12 but the file saved on my disc is different.
It is always shorter. Last few seconds are cropped. Usually it is 11:08 so I can't see last 4 seconds of the original video.
Is there any way to download full length file?

Video is hosted on Vimeo but it is paid content. It is available from https://happy-joga.pl
To access videos just type my email: pgdziarski on gmail domain.

When I check original video details with VDH it shows
```
bitrate 1614000
bulk undefined
chunked dash-adp
core_media {...}
created	1708078310531
descrPrefix Strumieniowanie DASH
duration 672.48 //__________which is 11 min 12s
extension mp4
...
```

There is no issue when I download videos from other public Vimeo (DASH streaming) or youtube  (HLS streaming).

I think there is slightly similar issue:
"Download ends before entire file downloaded" https://github.com/aclap-dev/video-downloadhelper/discussions/96

### Addon & CoApp version and details

VDH
version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: pl
license: accepted
key (only 16 first characters): 5616c7babb58470d
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\pgdzi",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by paulrouget** [comment]
Do you get the early ending on all variants? Because here it seems to end properly, but it seems that vimeo provides slight different end time depending on the selected size.

**Comment by GrampaWildWilly** [comment]
I visited the page whose URL you quoted in your original post.  At first, the VDH menu was empty.  But then I scrolled down a bit & found a video of duration 1:57.  After I hit play & let it play for about 1 second, the VDH menu populated with an array of choices.  VDH successfully downloaded the 1920x1080 variant but it was only 1:52.  Hmmmm.......

Scrolling the page down a bit further, I discovered a second video, this one of duration . . . well, it says 7:52 in the player, but the real duration happens to be 7:51, but I'm getting aheaed of myself.  After some page reloads & some Reloads of the VDH extension, I got VDH to show me variants for this second video.  They appeared to be for this second video because the variants on the VDH menu all reported duration 7:51.  But my attempts (multiple) to download the 1280x720 variant (the highest one offered for this one) ended with a Side Download Error & a stack trace that is of questionable use.

Time for the tried & true Plan B, which we used so often back in the Google forum.  I opened the Firefox Network Monitor & reloaded the page.  Nothing interesting showed up, so I did the trick of letting the first video play for 1 second & then paused it.  This caused a json with the word "master" in its name to show up.  First, I tried capturing it with Copy Response.  That got me something that did not contain anything I could use.  So then I grabbed the URL of the json & pasted it into my ffprobe invocation.  Then I looked for the string json within that URL & changed it to m3u8.  This is a trick we used to try when the other approach didn't work.  Yes, it's a bit of a guess, a stab in the dark, but it sometimes works.  It worked here.  Using the information provided by ffprobe, I downloaded the short video with ffmpeg & it was of duration 1:58, which I verified in VLC.

Then I scrolled the page down a bit until the second video came into view.  I did not even have to click the play button on this one.  A second json with "master" in its name appeared in the Network Monitor.  Going through the same process as for the short video, I once again downloaded this video with ffmpeg.  It was actually of duration 7:51, not the 7:52 claimed by the player on the web page.  I confirmed the duration 7:51 in VLC.

I have attached below a screenshot of the web page showing the Network Monitor with the second master json highlighted.  The first one is immediately above that one.  Look at some of the js items a bit higher up in the Network Monitor.  You'll note in the Domain column some mentions of Vimeo.  I have also attached the ffprobe reports & the ffmpeg logs for each video.  Note to Paul: Pay special attention to where I am executing ffprobe & ffmpeg from & their version numbers.

So I think this proves that VDH should be able to get these videos in the not too distant future, & get them with the correct duration.

Just as an aside . . . I did not sit & watch these videos all the way through, just sampled them at intervals.  But I happened to notice in the second video that one of the ladies is named Agnieszka.  I immediately liked her because one of my all-time favorite tennis players is Agnieszka Radwanska.  Too bad she's retired.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/28e25256-588d-44b5-a71d-5fca523ac157)
[ffprobe Yoga #1 Master.txt](https://github.com/aclap-dev/video-downloadhelper/files/14314370/ffprobe.Yoga.1.Master.txt)
[Yoga #1 mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14314372/Yoga.1.mp4.Log.txt)
[ffprobe Yoga #2 Master.txt](https://github.com/aclap-dev/video-downloadhelper/files/14314377/ffprobe.Yoga.2.Master.txt)
[Yoga #2 mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14314379/Yoga.2.mp4.Log.txt)

**Comment by Mattmjs** [comment]
I get the same result on that link with the first video of 1:58, the download finishes before it reaches 100%
On two resolutions I get 1:52

I see it happen on another : https://vimeo.com/251646742
It's 3:41, I get 3:36. It's ending the download before it has a chance to 100%

This might be the common factor, they're streaming using the chop segment format, not the dash format using **@ range=numbers**

![Screenshot 2024-02-17 150207](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/2ff22bf9-5d18-4c3c-94e8-96ed563d01d3)


![Screenshot 2024-02-17 150420](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/7cb8a418-291a-4975-96b3-0f69188f7993)

**Reply by paulrouget** [reply]
I confirm, I can reproduce locally. Thank you, I'll investigate.

**Reply by Gutdzilla** [reply]
Right, this is the same behavior. Thanks for commitment.

**Comment by GrampaWildWilly** [comment]
Retried with VDH 8.2.1.0a1 beta.  No change.

**Comment by paulrouget** [comment, marked as answer]
I think I found the issue.

Please give it a try: https://github.com/aclap-dev/video-downloadhelper/discussions/236

**Reply by Gutdzilla** [reply]
Great! That solves issue. Thanks a lot! :)

**Reply by adrianpebiker** [reply]
como se hace?

**Comment by Gutdzilla** [comment]
Great! I will check in a few hours

**Comment by adrianpebiker** [comment]
Hola, a mi me pasa igual y no se como hacerlo. descargo el video y dura unos 5 segundos menos. me podeis ayudar?

**Reply by GrampaWildWilly** [reply]
Click the Wiki link at the top of this page.  That takes you to the Wiki.  On the right side of that page there is a Table of Contents.  You want to look for 2 chapters: CoApp Installation, Beta versions.  Read what you find, follow the instructions, & install the latest releases of the software.  That is the VDH browser extension & the CoApp.  If you still have problems, post a proper problem report.  Those instructions are also in the Wiki in the chapter I need help.
